### PR TITLE
Add Helm Support

### PR DESCRIPTION
**Pull request by gavinfish**
- Install helm chart from file or from a helm chart repository
- Rollback a helm release

### CODE REVIEW

File: src/main/java/com/microsoft/jenkins/kubernetes/DeployType.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.microsoft.jenkins.kubernetes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum DeployType {
+    UNKNOWN("unknown"),
+    ORIGIN("origin"),
```
Comment (yungezz) [inline src/main/java/com/microsoft/jenkins/kubernetes/DeployType.java]: what's origin mean? better naming

File: src/main/java/com/microsoft/jenkins/kubernetes/KubernetesDeployContext.java [hunk 2]
```diff
@@ -64,6 +70,24 @@
     private ConfigFileCredentials kubeConfig;
     private TextCredentials textCredentials;
 
+    private DeployTypeClass deployTypeClass;
+    private String deployType;
+    private HelmContext helmContext;
+    private String helmChartLocation;
```
Comment (yungezz) [inline src/main/java/com/microsoft/jenkins/kubernetes/KubernetesDeployContext.java]: can below properties be  embedded into HelmContext?

File: src/main/java/com/microsoft/jenkins/kubernetes/KubernetesDeployContext.java [hunk 3]
```diff
@@ -82,10 +106,34 @@ public void configure(
...
+                            .withSingleCommand(HelmDeploymentCommand.class)
+                            .withStartCommand(HelmDeploymentCommand.class)
+                            .build();
+                } else if (Constants.HELM_COMMAND_TYPE_ROLLBACK.equals(commandType)) {
+                    commandService = CommandService.builder()
+                            .withSingleCommand(HelmRollbackCommand.class)
+                            .withStartCommand(HelmRollbackCommand.class)
+                            .build();
+                } else {
+                    throw new IOException("Unknown deploy type");
```
Comment (yuwzho) [inline src/main/java/com/microsoft/jenkins/kubernetes/KubernetesDeployContext.java]: IOException? Seems unsupported or argumentException more make sense.

Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/kubernetes/KubernetesDeployContext.java]: Done

File: src/main/java/com/microsoft/jenkins/kubernetes/DeployType.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    private static final Map<String, DeployType> LOOKUP = new HashMap<>();
+
+    static {
+        for (DeployType ref : DeployType.values()) {
+            LOOKUP.put(ref.getName(), ref);
+        }
+    }
+
+    public static DeployType get(String name) {
```
[the remark this answers is not in the document]
Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/kubernetes/DeployType.java]: `DeployType.valueOf(name)` can also implement this, just needs to take care case sensitive.

File: src/main/java/com/microsoft/jenkins/kubernetes/DeployType.java (hunk 1 above)
Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/kubernetes/DeployType.java]: Done

File: src/main/java/com/microsoft/jenkins/kubernetes/KubernetesDeployContext.java (hunk 2 above)
Comment (gavinfish) [reply src/main/java/com/microsoft/jenkins/kubernetes/KubernetesDeployContext.java]: The properties are in the HelmContext, here is just for UI reflection.

**Comment by ozbillwang** [comment]
@gavinfish 

could I ask for help?

Since this PR is not approved and merged, I want to use this plugin to manage Helm chart deployment. 

I can write some scripts to cover the deployment steps first, until this new feature is ready. 

But I faced the problem at beginning, that I don't know how to use the Binding kubeconfig credential (KUBECONIFG)

I raised the issue in #137 , maybe you can help me directly?

**Comment by ozbillwang** [comment]
@gavinfish 

What's the reason to close it directly? Will we still plan to merge this change?

**Comment by gavinfish** [comment]
@ozbillwang I am not working on this any more. Feel free to pick up.

**Pull request closed without merging**